### ISSUE TITLE
[Accessibility]: Navigating back after clicking on a search result doesn't restore focus 

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34724]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

After clicking on a search result and then going back, focus is not restored to the search result that was clicked, causing keyboard users to have to find the result in the list all over again.

### Expected behavior

Focus should be restored to the search result that was clicked

### Additional details

@sourcegraph/frontend-platform: Is this an issue application-wide? Is this being tracked by FPT in a more general way?

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by umpox** [comment]
@limitedmage We can look into this.

My initial thought is that we shouldn't restore focus as we can't be fully sure that the user wants that to happen (they might just want to access something else on the page). In that case, it might be better to keep the default focus for the page and let them reliably navigate to their relevant area (generally still quite fast through landmarks + headings).

Not 100% so will look further!

**Comment by felixfbecker** [comment]
General browser behavior is to restore scroll position, but not focus. E.g. if you focus and activate a link on Wikipedia, then go back, `<body>` will be `document.activeElement`, not the link that was activated before. I think that's the behavior we should probably aim for too. At least it's not a WCAG violation. We do support scroll position restoration from a quick test, so I'm closing this.
